prometheus/macros/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+extern crate proc_macro2;
```

Comment:
Cool idea to make a macro for this! It’s a nice approach on the definition side, and the macro itself looks pretty simple too, nice.